src/Spectre.Service/Savers/DatasetSaver.cs

```diff
@@ -0,0 +1,105 @@
+﻿/*
+ * DatasetSaver.cs
+ * Class saving datasets from either local cache or memory.
+ *
+   Copyright 2017 Dariusz Kuchta
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using Spectre.Data.Datasets;
+using Spectre.Dependencies;
+using Spectre.Service.Configuration;
+using Spectre.Service.Loaders;
+
+namespace Spectre.Service.Savers
+{
+    /// <summary>
+    /// Class saving datasets from either local cache or memory.
+    /// </summary>
+    public class DatasetSaver
+    {
+        #region Fields
+
+        /// <summary>
+        /// Root for cache directory.
+        /// </summary>
+        private readonly string _cacheRoot;
+
+        /// <summary>
+        /// Root for remote directory.
+        /// </summary>
+        private readonly string _remoteRoot;
+
+        #endregion
+
+        #region Constructor
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DatasetSaver"/> class.
+        /// </summary>
+        /// <param name="dataRootConfig">Validated configuration of needed directories.</param>
+        public DatasetSaver(DataRootConfig dataRootConfig)
+        {
+            _cacheRoot = dataRootConfig.LocalPath;
+            _remoteRoot = dataRootConfig.RemotePath;
+
+            FileSystem = DependencyResolver.GetService<IFileSystem>();
+        }
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// Handle to file system.
+        /// </summary>
+        private IFileSystem FileSystem { get; }
```

Comment:
Can't it be `readonly`?

Reply:
Without 'set' the property is implicitly read-only.